src/parser/mod.rs

```diff
@@ -92,34 +99,85 @@ impl UserAgentParser {
     /// ```
     pub fn from_bytes(bytes: &[u8]) -> Result<UserAgentParser, Error> {
         let regex_file: RegexFile = serde_yaml::from_slice(bytes)?;
-        UserAgentParser::try_from(regex_file)
+        Self::try_from(regex_file)
+    }
+
+    fn _build_from_bytes(
+        bytes: &[u8],
+        builder: UserAgentParserBuilder,
+    ) -> Result<UserAgentParser, Error> {
+        let regex_file: RegexFile = serde_yaml::from_slice(bytes)?;
+        Self::_try_from(
+            regex_file,
+            builder.device,
+            builder.os,
+            builder.user_agent,
+            builder.unicode,
+        )
     }
 
     /// Attempts to construct a `UserAgentParser` from a reference to an open
     /// `File`. This `File` should be a the `regexes.yaml` depended on by
     /// all the various implementations of the UA Parser library.
     pub fn from_file(file: std::fs::File) -> Result<UserAgentParser, Error> {
         let regex_file: RegexFile = serde_yaml::from_reader(file)?;
-        UserAgentParser::try_from(regex_file)
+        Self::try_from(regex_file)
+    }
+
+    fn _build_from_file(
+        file: std::fs::File,
+        builder: UserAgentParserBuilder,
+    ) -> Result<UserAgentParser, Error> {
+        let regex_file: RegexFile = serde_yaml::from_reader(file)?;
+        Self::_try_from(
+            regex_file,
+            builder.device,
+            builder.os,
+            builder.user_agent,
+            builder.unicode,
+        )
     }
 
     pub fn try_from(regex_file: RegexFile) -> Result<UserAgentParser, Error> {
-        let mut device_matchers = Vec::with_capacity(regex_file.device_parsers.len());
-        let mut os_matchers = Vec::with_capacity(regex_file.os_parsers.len());
-        let mut user_agent_matchers =
-            Vec::with_capacity(regex_file.user_agent_parsers.len());
+        Self::_try_from(regex_file, true, true, true, true)
+    }
 
-        for parser in regex_file.device_parsers {
-            device_matchers.push(device::Matcher::try_from(parser)?);
-        }
+    fn _try_from(
+        regex_file: RegexFile,
+        device: bool,
+        os: bool,
+        user_agent: bool,
+        unicode: bool,
+    ) -> Result<UserAgentParser, Error> {
+        let device_matchers = if device {
```

Comment:
I explored a few options on how to disable/enable a specific class of parser. Using crate features was neat, but felt overkill for something that could just be provided as config. The most correct approach is probably making the `Vec<device::Matcher>` an `Option<Vec<device::Matcher>>` and making Client have optional fields for all the matches, but that was a bigger footprint and I wasn't sure if that was worth it. An empty Vec is what we're getting now with our stripped down regex yaml file and it works well enough.